test-requirements.txt

```diff
@@ -1,3 +1,4 @@
+pyarrow>=4.0.0
```

Comment:
why is this required? if/when this is missing in an install of the provider, will the conditional pyspark logic still work?

Reply:
Yeah, it specifically fails for spark connect flows 😞

Reply:
ok, just to be clear, this is only adding it to test deps. So to confirm, this is just required to set up a spark connect dataframe, not required for our business logic?

Reply:
thanks 🙇